Handle the caches update correctly for retracted patches in CLM

hustodemon opened the pull request:
Make sure the following caches are aligned correctly when dealing with retracted patches:

- `rhnServerNeeded`
- `rhnChannelNewestPackage`

The changes are also relevant for non-CLM scenarios.

Review commit-by-commit.

TODO
- [x] DB Migration

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: more or less a bugfix :smile:

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)


## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

mcalmer commented on java/code/src/com/redhat/rhn/manager/errata/cache/test/RetractedPatchesCacheManagerTest.java:
```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2009--2014 Red Hat, Inc.
```
A new file should get the correct copyright

mcalmer reviewed (approved):
Just the copyright header needs to get changed. The code looks good

lucidd commented on java/code/src/com/redhat/rhn/domain/server/ServerFactory.java:
```diff
@@ -261,6 +261,20 @@ public static List lookupServersWithCustomKey(Long userId, Long cikid) {
+     *
+     * @param serverId the server ID
+     */
+    public static void updateServerNeededCache(Long serverId) {
```
i'm not sure but it does not look like this expects `null` to be passed so i would just keep this `long`.

hustodemon replied:
Makes sense. Changing.

hustodemon commented:
Addressed your comments and added the migration (I've used `4.2.5 -> 4.2.6` directory for now. Before merging to `master`, we should move our migration to the up-to-date dir).

Thank you for your reviews!
